miasm2/arch/aarch64/arch.py

```
@@ -0,0 +1,1968 @@
+
+    def __init__(self, *args, **kargs):
+        super(instruction_aarch64, self).__init__(*args, **kargs)
+
+    @staticmethod
+    def arg2str(e, pos=None):
+        wb = False
+        if isinstance(e, m2_expr.ExprId) or isinstance(e, m2_expr.ExprInt):
+            return str(e)
+        elif isinstance(e, m2_expr.ExprOp) and e.op in shift_expr:
+            op_str = shift_str[shift_expr.index(e.op)]
+            return "%s %s %s" % (e.args[0], op_str, e.args[1])
+        elif isinstance(e, m2_expr.ExprOp) and e.op == "slice_at":
+            return "%s LSL %s" % (e.args[0], e.args[1])
+        elif isinstance(e, m2_expr.ExprOp) and e.op in extend_lst:
+            op_str = e.op
+            return "%s %s %s" % (e.args[0], op_str, e.args[1])
+        elif isinstance(e, m2_expr.ExprOp) and e.op == "postinc":
+            if e.args[1].arg != 0:
+                return "[%s], %s" % (e.args[0], e.args[1])
+            else:
+                return "[%s]" % (e.args[0])
+        elif isinstance(e, m2_expr.ExprOp) and e.op == "preinc_wb":
+            if e.args[1].arg != 0:
+                return "[%s, %s]!" % (e.args[0], e.args[1])
+            else:
+                return "[%s]" % (e.args[0])
+        elif isinstance(e, m2_expr.ExprOp) and e.op == "preinc":
+            if len(e.args) == 1:
+                return "[%s]" % (e.args[0])
+            elif not isinstance(e.args[1], m2_expr.ExprInt) or e.args[1].arg != 0:
+                return "[%s, %s]" % (e.args[0], e.args[1])
+            else:
+                return "[%s]" % (e.args[0])
+        elif isinstance(e, m2_expr.ExprOp) and e.op == 'segm':
+            arg = e.args[1]
+            if isinstance(arg, m2_expr.ExprId):
+                arg = str(arg)
+            elif arg.op == 'LSL' and arg.args[1].arg == 0:
+                arg = str(arg.args[0])
+            else:
+                arg = "%s %s %s" % (arg.args[0], arg.op, arg.args[1])
+            return '[%s, %s]' % (e.args[0], arg)
+
+        fds
```

You should raise an error instead